Source/Provers/SMTLib/SMTLibInteractiveTheoremProver.cs

```diff
@@ -41,6 +41,19 @@ public override Task GoBackToIdle()
       return Process.PingPong();
     }
 
+    public void ClearProverErrors()
+    {
+      lock (proverErrors)
+      {
+        if (proverErrors.Count > 0)
+        {
+          proverWarnings.RemoveAll(x => true);
+          proverErrors.RemoveAll(x => true);
+          SetupProcess();
```

Comment:
Would it make sense to, instead, set `processNeedsRestart` to `true` here so that it gets restart in `PossiblyRestart` which is called a few lines below? That would guarantee that the restarted process would always receive the text in `common`.